.github/workflows/propagateUserHome.gradle.kts

```diff
@@ -0,0 +1,26 @@
+/**
+ * Gradle init script to correct an unset 'user.name' Java system property resulting from the defect
+ * described in https://bugs.openjdk.java.net/browse/JDK-8193433.
+ */
+val userHome: String? = System.getProperty("user.home")
+if (userHome.isNullOrEmpty() || userHome.equals("?")) {
+    val home = System.getenv("HOME")
+    if (home.isNullOrEmpty()) {
+        throw IllegalStateException("environment variable \"HOME\" must be set when user.home='${userHome}'")
+    }
+
+    /*
+     * Replace Java's broken user.home value in both this JVM and for each test JVM.
+     */
+    println("Setting user.home=\"${home}\"")
+    System.setProperty("user.home", home)
+
+    gradle.allprojects {
+        tasks.withType<Test> {
```

Comment:
You might be able to use:
```kotlin
tasks.withType<JavaForkOptions> {
}
```
which will catch both `Test` and `JavaExec` tasks.

Reply:
I'll try it and see what happens.